src/settings.h

```diff
@@ -196,6 +197,27 @@ class Settings
         ms_appVersion = version;
     }
 
+	/// Adds an additional header parameter.
+	static void AddHttpHeader(const char *name, const char *value)
+	{
+		CriticalSectionLocker lock(ms_csVars);
+		ms_httpHeaders += std::string(name) + std::string(": ") + std::string(value) + std::string("\r\n");
```

Comment:
I don't believe, creating temporary objects is necessary here:
for `std::string(": ")` and `std::string("\r\n")` SSO will do its job, but for `std::string(value)` it will most likely lead to memory allocation and then deallocation

Reply:
More importantly, and avoiding this issue, it would make more sense to have `ms_httpHeaders` as `map<string, string>`, to allow overriding and to make duplicate headers impossible...